example/js/lib/api/diplomat-runtime.js

```diff
@@ -80,8 +80,12 @@ export class DiplomatBuf {
     this.ptr = ptr;
     this.size = size;
     this.free = () => {
-      wasm.diplomat_free(this.ptr, this.size, align);
-      DiplomatBuf_finalizer.unregister(this);
+      const successfully_unregistered = DiplomatBuf_finalizer.unregister(this);
```

Comment:
nit: i think we shoudl have a function in DiplomatRuntime that can handle this

Reply:
This is the only place where we ever unregister, I think this should be fine? I also don't anticipate we add anything else where we might want to manually free

Reply:
Yeah my reasoning is more so that the generated code can be as compact as possible

Reply:
`this.free = make_buf_unregisterer(this)` or something